core/src/block.rs

```diff
@@ -190,8 +192,17 @@ impl OpenBlock {
         parent_block_number: BlockNumber,
         parent_block_timestamp: u64,
     ) -> Result<(), Error> {
-        for tx in transactions {
-            self.push_transaction(tx.clone(), client, parent_block_number, parent_block_timestamp)?;
+        for (index, tx) in transactions.iter().enumerate() {
+            self.push_transaction(
+                tx.clone(),
+                client,
+                parent_block_number,
+                parent_block_timestamp,
+                TransactionLocation {
```

Comment:
I wonder whether we need to create the transaction location early here.
Only `self_nomiate` needs it and the parent block number is already passed.
How about passing the transaction index here and creating the location when calling the function?

Reply:
I want to distinguish TransactionLocation and parent_block_number.
In the RPC method "chain_executeTransaction" does not have TransactionLocation but can have parent_block_number.

However, I found that Foundry doesn't have **chain_executeTransaction** (Foundry had it before) in Foundry. If we can ensure that all transactions are executed in a block, and always has an index, I agree with you.

Do you know whether the transaction always has transaction index or not?

Reply:
We removed `chain_executeTransaction`, but it will re-implemented with the module system.
So I cannot say that all transaction is executed in the block, but I think it can be bypassed by adding the transaction index as an argument of the RPC.

Reply:
or how about to fix the transaction index as zero when the transaction is executed by the RPC?

Reply:
@sgkim126 
Passing index or using zero looks reasonable.
I'll fix this PR like followings:
* Use TransactionLocation only when there is no parent block number.
* Use TransactionIndex instead of Option<TransactionLocation>